tests/lambda_endpoints/conftest.py

```diff
@@ -84,15 +86,31 @@ def events_client():
         yield boto3.client("events", region_name="us-west-2")
 
 
+# Check if postgres is available on the system. If it is returncode == 0
+POSTGRES_AVAILABLE = (
+    subprocess.run("which psql", shell=True, check=False).returncode == 0
+)
+
+if POSTGRES_AVAILABLE:
+
+    @pytest.fixture()
+    def connection(postgresql):
+        """Use a postgres connection string."""
+        return f"postgresql+psycopg://{postgresql.info.user}:@{postgresql.info.host}:{postgresql.info.port}/{postgresql.info.dbname}"
```

Comment:
Are these, user and host and port and so on, something we need to set locally for test or does it know it if we have `psql`?

Reply:
These are available from the `pytest-postgresql` test fixture called on entry to this fixture `connection(postgresql)` gives us that postgresql information for us (it spins up a DB in the background and spits out the specific connection information for us).

This won't get used if we don't have `psql` installed.

Reply:
good to know. Thank you!